src/Spectre.libPlatform/Math.h

```diff
@@ -0,0 +1,354 @@
+/*
+* Math.h
+* Basic mathematical operations on vectors.
+*
+Copyright 2017 Grzegorz Mrukwa
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+#include <vector>
+#include <span.h>
+#include "Transform.h"
+
+namespace Spectre::libPlatform::Math
+{
+// @gmrukwa: vector vs vector
+
+/// <summary>
+/// Add two vectors elementwise.
+/// </summary>
+/// <param name="first">The first.</param>
+/// <param name="second">The second.</param>
+/// <returns>Vector, elements of which are sums of corresponding elements in source vectors.</returns>
+template< class DataType >
+constexpr std::vector<DataType> plus(gsl::span<const DataType> first, gsl::span<const DataType> second)
```

Comment:
Right now, they're by means of std::transform. It is *as good as loops are* [source](https://www.youtube.com/watch?v=27tFyKc6g3U). Therefore, until we have some overpowered-low-level-black-magic-with-ultra-special-buffs, I'd stick to this. Either way - it's a single place for a substitution ;)